src/sparse_blas/backends/cusparse/operations/cusparse_spsv.cpp

```diff
@@ -0,0 +1,249 @@
+        auto cu_x = x_handle->backend_handle;
+        auto cu_y = y_handle->backend_handle;
+        auto type = A_handle->value_container.data_type;
+        set_matrix_attributes(__func__, cu_a, A_view);
+        auto cu_op = get_cuda_operation(type, opA);
+        auto cu_type = get_cuda_value_type(type);
+        auto cu_alg = get_cuda_spsv_alg(alg);
+        auto cu_descr = spsv_descr->cu_descr;
+        set_pointer_mode(cu_handle, is_alpha_host_accessible);
+        auto status = cusparseSpSV_bufferSize(cu_handle, cu_op, alpha, cu_a, cu_x, cu_y, cu_type,
+                                              cu_alg, cu_descr, &temp_buffer_size);
+        check_status(status, __func__);
+    };
+    auto event = dispatch_submit(__func__, queue, functor, A_handle, x_handle, y_handle);
+    event.wait_and_throw();
+    spsv_descr->buffer_size_called = true;
+}
+
+inline void common_spsv_optimize(oneapi::mkl::transpose opA, bool is_alpha_host_accessible,
+                                 matrix_view A_view, matrix_handle_t A_handle,
+                                 dense_vector_handle_t x_handle, dense_vector_handle_t y_handle,
+                                 spsv_alg alg, spsv_descr_t spsv_descr) {
+    check_valid_spsv("spsv_optimize", A_view, A_handle, x_handle, y_handle,
+                     is_alpha_host_accessible);
+    if (!spsv_descr->buffer_size_called) {
+        throw mkl::uninitialized("sparse_blas", "spsv_optimize",
+                                 "spsv_buffer_size must be called before spsv_optimize.");
+    }
+    spsv_descr->optimized_called = true;
+    spsv_descr->last_optimized_opA = opA;
+    spsv_descr->last_optimized_A_view = A_view;
+    spsv_descr->last_optimized_A_handle = A_handle;
+    spsv_descr->last_optimized_x_handle = x_handle;
+    spsv_descr->last_optimized_y_handle = y_handle;
+    spsv_descr->last_optimized_alg = alg;
+}
+
+void spsv_optimize_impl(cusparseHandle_t cu_handle, oneapi::mkl::transpose opA, const void *alpha,
+                        matrix_view A_view, matrix_handle_t A_handle,
+                        dense_vector_handle_t x_handle, dense_vector_handle_t y_handle,
+                        spsv_alg alg, spsv_descr_t spsv_descr, void *workspace_ptr,
+                        bool is_alpha_host_accessible) {
+    auto cu_a = A_handle->backend_handle;
+    auto cu_x = x_handle->backend_handle;
+    auto cu_y = y_handle->backend_handle;
+    auto type = A_handle->value_container.data_type;
+    set_matrix_attributes("optimize_spsv", cu_a, A_view);
+    auto cu_op = get_cuda_operation(type, opA);
+    auto cu_type = get_cuda_value_type(type);
+    auto cu_alg = get_cuda_spsv_alg(alg);
+    auto cu_descr = spsv_descr->cu_descr;
+    set_pointer_mode(cu_handle, is_alpha_host_accessible);
+    auto status = cusparseSpSV_analysis(cu_handle, cu_op, alpha, cu_a, cu_x, cu_y, cu_type, cu_alg,
+                                        cu_descr, workspace_ptr);
+    check_status(status, "optimize_spsv");
```

Comment:
```suggestion
    check_status(status, "spsv_optimize");
```

Reply:
Fixed in https://github.com/oneapi-src/oneMKL/pull/527/commits/325f79495c2090b95538e0eaaa80ccf40f2e9737